### PR TITLE
[Hotfix] Domain extraction management command

### PR DESCRIPTION
**Pull request by jwalz**
## Purpose
Extract NotableDomains and DomainReferences from OSF resources pre-dating the latest release

## Changes
* Add a management command for extracting domains and domain references
* Homogenize interfaces around getting spam check content
* Improve handling of spammy comments

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

**Comment by Johnetordoff** [comment]
@jwalz BTW one thing that Blaine mentioned is that user's personal sites still aren't domained checked, you might want to shove that in here.

File: osf/management/commands/backfill_domain_references.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            type=bool,
+            default=False,
+            help='Run queries but do not create instances',
+        )
+        parser.add_argument(
+            '--model_name',
+            type=str,
+            required=True,
+            help='The name of the model to be searched for domains, '
+                 'remember WikiVersions app label name is addons_wiki.WikiVersion',
```
Comment (mfraezz) [inline osf/management/commands/backfill_domain_references.py:63]: Could/should this have [choices](https://docs.python.org/3/library/argparse.html#choices) specified?

Also, `help` implies that `WikiVersion` is a valid model, but would raise an `AttributeError` due to `model.SPAM_CHECK_FIELDS` on line 23.

File: osf/management/commands/backfill_domain_references.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from django.db.models import Q, OuterRef, Exists
+from django.apps import apps
+from django.core.management.base import BaseCommand
+from osf.external.spam import tasks as spam_tasks
+from osf.models import DomainReference
+
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_SEARCH_REGEX = r'(\w+\.\w+\w+\.\w+)|\w+\.\w+\w+'
```
Comment (mfraezz) [inline osf/management/commands/backfill_domain_references.py:15]: Should this instead use `osf.external.spam.tasks.DOMAIN_REGEX`?

File: osf/management/commands/backfill_domain_references.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from django.contrib.contenttypes.models import ContentType
+from osf.models import Node
+
+
+def backfill_domain_references(model_name, dry_run=False, batch_size=None):
+    model = apps.get_model(model_name)
+
+    search_fields = list(model.SPAM_CHECK_FIELDS)
+    if model == Node:
```
Comment (mfraezz) [inline osf/management/commands/backfill_domain_references.py:24]: If we want to check `Registration` wikis too: `if issubclass(model, AbstractNode):`

File: osf/management/commands/backfill_domain_references.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    spam_content_query = reduce(operator.or_, spam_queries)
+
+    spam_check_items = model.objects.annotate(
+        exclude=Exists(
+            DomainReference.objects.filter(
+                referrer_content_type=ContentType.objects.get_for_model(model),
+                referrer_object_id=OuterRef('id')
+            )
+        )
+    ).filter(exclude=False).filter(spam_content_query)[:batch_size]
```
Comment (mfraezz) [inline osf/management/commands/backfill_domain_references.py:37]: Should this use `.order_by('?')` to mitigate the chances of locking on a batch with objects that have no domains to reference?

**Comment by jwalz** [comment]
Closed in favor of #10353

**Pull request closed without merging**